### PR TITLE
feat: Migrate to TypeScript, build with tsdx and update Storybook

### LINKED ISSUE #44: Missing props in SemmaticDatepickerProps interface definition
# ❔ Question
I am working on a project written by typescript and I would like to use ReadOnly props in SemmaticDatepicker, then I got compile error because readOnly is not defined in interface SemanticDatepickerProps. Following codes are from "react-semantic-ui-datepickers/dist/index.d.ts"

```
export interface SemanticDatepickerProps {
  clearable?: boolean;
  date?: Date;
  disabled?: boolean;
  error?: boolean;
  firstDayOfWeek?: number;
  format?: string;
  icon?: SemanticICONS;
  iconPosition?: 'left';
  id?: string;
  keepOpenOnClear?: boolean;
  keepOpenOnSelect?: boolean;
  label?: string;
  loading?: boolean;
  locale?: Locale;
  maxDate?: Date;
  minDate?: Date;
  name?: string;
  onDateChange: (newDate: Date | Date[] | null) => void;
  placeholder?: string;
  pointing: 'left' | 'right' | 'top left' | 'top right';
  selected?: Date | Date[];
  showOutsideDays?: boolean;
  size?: SemanticSIZES;
  transparent?: boolean;
  type: 'basic' | 'range';
}
```

### PR DESCRIPTION
**Pull request by arthurdenner**
**What kind of change does this PR introduce?**
Feature, bug fix and docs update. Closes #44.

**What is the current behavior?**
- Typings are maintained manually which is not great;
- Our build is not so small, around [~63kb](https://bundlephobia.com/result?p=react-semantic-ui-datepickers@1.11.1);
- Our documentation is not up-to date and has some invalid examples.

**What is the new behavior?**
- The library uses TypeScript and the typing are generated automatically;
- The build size with tsdx is smaller;
- Our documentation uses a recent version of Storybook and only has valid examples.

**Checklist**:

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged 

:warning: **There are a few things left in order to merge this PR:** :warning: 

- [x] ~There is only one CSS file after building the project and it's tagged as development;~
- [ ] ~We are using an old version of rollup-plugin-copy, would be nice if we could update it.~ (it's better to skip it for now since the latest version is not compatible with tsdx)

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    keepOpenOnSelect: boolean;
+    locale: Locale;
+    onBlur: (event?: React.SyntheticEvent) => void;
+    onDateChange: (date: Date | Date[] | null) => void;
+    pointing: 'left' | 'right' | 'top left' | 'top right';
+    type: 'basic' | 'range';
+  };
+
+export type DayzedProps = {
+  children: (children: any) => void;
```
Comment (emersonlaurentino) [inline src/types/index.ts]: same

File: src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import SemanticDatepicker from './components/datepicker';
+
+export default SemanticDatepicker;
```
Comment (emersonlaurentino) [inline src/index.tsx]: you can write like this

`export { default }  from './components/datepicker';`

File: src/types/index.ts (hunk 1 above)
Comment (arthurdenner) [reply src/types/index.ts]: In this case, since `children` is a function, we need to keep this declaration.

**Pull request merged**